bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageImpl.java

```diff
@@ -197,4 +201,32 @@ public boolean hasCloudconfigSupport() {
         }
         return hasCloudconfigSupport;
     }
+
+    @Override
+    public String getMainContentSelector() {
+        PageManager pageManager = request.getResourceResolver().adaptTo(PageManager.class);
```

Comment:
I would cache the selector in an instance variable, just in case somebody decides to call this method twice for the same model instance:

```
if (mainContentSelector == null) {
    .....;
    mainContentSelector = ...;
}
return mainContentSelector;
```